src/containers/Application/ApplicationPage.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react'
+import { FullStructure, ResponsesByCode } from '../../utils/types'
+import useGetFullApplicationStructure from '../../utils/hooks/useGetFullApplicationStructure'
+import { ApplicationStatus } from '../../utils/generated/graphql'
+import { useApplicationState } from '../../contexts/ApplicationState'
+import { useUserState } from '../../contexts/UserState'
+import { useRouter } from '../../utils/hooks/useRouter'
+import { Loading, NoMatch } from '../../components'
+import strings from '../../utils/constants'
+import messages from '../../utils/messages'
+import { Button, Grid, Header, Message, Segment, Sticky } from 'semantic-ui-react'
+
+interface ApplicationProps {
+  structure: FullStructure
+  responses?: ResponsesByCode
+}
+
+const ApplicationPage: React.FC<ApplicationProps> = ({ structure }) => {
+  const [isStrictPage, setIsStrictPage] = useState(null)
+  const { error, isLoading, fullStructure, responsesByCode } = useGetFullApplicationStructure(
+    structure
+  )
+  const {
+    userState: { currentUser },
+  } = useUserState()
+  const { push } = useRouter()
+
+  console.log('Structure', fullStructure)
+
+  useEffect(() => {
+    if (!structure) return
+
+    // Re-direct based on application status and progress
+    if (structure.info.current?.status === ApplicationStatus.ChangesRequired)
+      push(`/applicationNEW/${structure.info.serial}`)
+    if (structure.info.current?.status !== ApplicationStatus.Draft)
+      push(`/applicationNEW/${structure.info.serial}/summary`)
+
+    // TO-DO: Redirect based on Progress (wait till Progress calculation is done)
+  }, [structure])
+
+  if (error) return <Message error header={strings.ERROR_APPLICATION_PAGE} list={[error]} />
+  if (isLoading) return <Loading />
+
```

Comment:
Shouldn't have a check for `fullStructure` not `undefined` here?

Reply:
With that you wouldn't need to cast to FullStructure type on the render block

Reply:
Yes, have done that in the next PR #347